src/main/java/io/datanerds/avropatch/Patch.java

```diff
@@ -5,39 +5,127 @@
 import org.apache.avro.reflect.AvroSchema;
 
 import javax.annotation.Generated;
+import javax.annotation.concurrent.ThreadSafe;
 import java.util.*;
+import java.util.stream.Stream;
 
 /**
- * This class represents a JSON PATCH operation holding a sequence of operations to apply to a given object.
+ * This class represents a JSON PATCH operation holding a sequence of operations to apply to a given object. It provides
+ * shortcuts for accessing the list of operations and some additional fields for metadata information, such as the
+ * resource identifier, a timestamp and a {@link Map} holding arbitrary header information.
  *
  * @see <a href="https://tools.ietf.org/html/rfc6902">https://tools.ietf.org/html/rfc6902</a>
  */
-public class Patch {
+@ThreadSafe
+public class Patch<T> {
 
     @AvroSchema(DefaultSchema.HEADERS)
-    private final Map<String, ?> headers;
+    private final Map<String, ? super Object> headers;
+    @AvroSchema(DefaultSchema.VALUE)
+    private final T resource;
+    @AvroSchema(DefaultSchema.TIMESTAMP)
+    private final Date timestamp;
     private final List<Operation> operations;
 
     @SuppressWarnings("unused") // no-arg constructor needed by Avro
     private Patch() {
-        this(new ArrayList<>(), new HashMap<>());
+        this(null, Collections.emptyList());
```

Comment:
this(null) should be sufficient as we have a constructor with only resource as the argument.